# Proposta: Projeto Final


### Escolher uma fonte de dados
- API
- Arquivo  ✅

### Objetivos:

1. Trabalhar na qualidade dos dados
   - Utilizar `ydata_profiling` para gerar relatórios de perfil dos dados. ✅
2. Implementar as expectativas com a biblioteca Great Expectations.
3. Utilizar DBT para a transformação dos dados.
4. Criar camadas de dados - Gravando no Postgres - Bronze, Silver e Gold.
5. Trabalhar na limpeza dos dados.


In [ ]:
pip install ydata_profiling

In [3]:
from ydata_profiling import ProfileReport

ModuleNotFoundError: No module named 'ydata_profiling'

In [ ]:
url = "http://data.insideairbnb.com/brazil/rj/rio-de-janeiro/2023-12-26/visualisations/listings.csv"
#DOC https://docs.google.com/spreadsheets/d/1iWCNJcSutYqpULSQHlNyGInUvHg2BoUGoNRIGa6Szc4/edit#gid=1322284596

df_listings = pd.read_csv(url) #ler csv da url

#print(df_listings.head(1))
print(df_listings.describe())

                 id       host_id  neighbourhood_group      latitude  \
count  3.600800e+04  3.600800e+04                  0.0  36008.000000   
mean   4.445730e+17  1.784179e+08                  NaN    -22.967580   
std    4.418582e+17  1.769684e+08                  NaN      0.034687   
min    1.787800e+04  1.671000e+03                  NaN    -23.073276   
25%    2.863794e+07  2.416786e+07                  NaN    -22.984712   
50%    5.761368e+17  9.644573e+07                  NaN    -22.972860   
75%    8.726761e+17  3.256093e+08                  NaN    -22.956277   
max    1.053823e+18  5.527145e+08                  NaN    -22.749690   

          longitude          price  minimum_nights  number_of_reviews  \
count  36008.000000   34237.000000    36008.000000       36008.000000   
mean     -43.249377    1211.719572        4.419351          19.545545   
std        0.099241    5790.937363       22.738877          40.600380   
min      -43.723009       0.000000        1.000000         

In [ ]:
from pandas_profiling import ProfileReport

# Criar o relatório
profile = ProfileReport(df_listings, title="Pandas Profiling Report")

# Caminho de saída para o arquivo HTML dentro do Colab [Clicar no ícone de pasta para ver o arquivo]
output_path = '/content/listings_profiling.html'

# Salvar os resultados em um arquivo HTML no ambiente do Colab
profile.to_file(output_path)


/usr/local/lib/python3.10/dist-packages/ydata_profiling/profile_report.py:354: UserWarning: Try running command: 'pip install --upgrade Pillow' to avoid ValueError
  warnings.warn(


Summarize dataset:   0%|          | 0/5 [00:00<?, ?it/s]

Generate report structure:   0%|          | 0/1 [00:00<?, ?it/s]

Render HTML:   0%|          | 0/1 [00:00<?, ?it/s]

Export report to file:   0%|          | 0/1 [00:00<?, ?it/s]

## Alerts gerados pelo ProfileReport

- **latitude** está altamente correlacionada com **longitude** - Alta correlação
- **longitude** está altamente correlacionada com **latitude** - Alta correlação
- **number_of_reviews** está altamente correlacionada com **number_of_reviews_ltm** e 1 outro campo - Alta correlação
- **number_of_reviews_ltm** está altamente correlacionada com **number_of_reviews** e 1 outro campo - Alta correlação
- **reviews_per_month** está altamente correlacionada com **number_of_reviews** e 1 outro campo - Alta correlação
- **room_type** está altamente desbalanceada (58.4%) - Desbalanceada
- **neighbourhood_group** possui 36,008 (100.0%) valores faltantes - Faltantes
- **price** possui 1,771 (4.9%) valores faltantes - Faltantes
- **last_review** possui 10,302 (28.6%) valores faltantes - Faltantes
- **reviews_per_month** possui 10,302 (28.6%) valores faltantes - Faltantes
- **license** possui 36,008 (100.0%) valores faltantes - Faltantes
- **price** está altamente inclinada (γ1 = 66.96707573) - Inclinada
- **minimum_nights** está altamente inclinada (γ1 = 28.22092657) - Inclinada
- **id** possui valores únicos - Únicos
- **neighbourhood_group** é um tipo não suportado, verifique se precisa de limpeza ou análise adicional - Não suportado
- **license** é um tipo não suportado, verifique se precisa de limpeza ou análise adicional - Não suportado
- **number_of_reviews** possui 10,302 (28.6%) zeros - Zeros
- **availability_365** possui 5,475 (15.2%) zeros - Zeros
- **number_of_reviews_ltm** possui 14,214 (39.5%) zeros - Zeros


In [ ]:
!pip install ipython-sql
!pip install sqlalchemy
!pip install psycopg2